src/domain/event/eventInfo/__tests__/OtherEventTimes.test.tsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+
+import translations from '../../../../common/translation/i18n/fi.json';
+import {
+  EventFieldsFragment,
+  EventListDocument,
+} from '../../../../generated/graphql';
+import getDateRangeStr from '../../../../util/getDateRangeStr';
+import { fakeEvent, fakeEvents } from '../../../../util/mockDataUtils';
+import { render, screen, userEvent, waitFor } from '../../../../util/testUtils';
+import OtherEventTimesEvents from '../OtherEventTimes';
+
+const startTime = '2020-10-01T16:00:00Z';
+const endTime = '2020-10-01T18:00:00Z';
+
+const superEventId = 'hel:123';
+const superEventInternalId = `https://api.hel.fi/linkedevents/v1/event/${superEventId}`;
+
+const mockEvent = fakeEvent({
+  superEvent: { internalId: superEventInternalId },
+}) as EventFieldsFragment;
+
+const meta = {
+  count: 20,
+  next:
+    'https://api.hel.fi/linkedevents/v1/event/?include=keyword,location&page=2&sort=start_time&start=2020-08-11T03&super_event=hel:123',
+  previous: null,
+  __typename: 'Meta',
+};
+
+const otherEventsResponse = {
+  data: {
+    eventList: {
+      ...fakeEvents(
+        10,
+        range(1, 11).map((i) => ({
+          endTime: addDays(new Date(endTime), i).toISOString(),
+          startTime: addDays(new Date(startTime), i).toISOString(),
+        }))
+      ),
+      meta,
+    },
+  },
+};
+
+const otherEventsLoadMoreResponse = {
+  data: {
+    eventList: {
+      ...fakeEvents(
+        10,
+        range(11, 21).map((i) => ({
+          endTime: addDays(new Date(endTime), i).toISOString(),
+          startTime: addDays(new Date(startTime), i).toISOString(),
+        }))
+      ),
+      meta: { ...meta, next: null },
+    },
+  },
+};
+
+const variables = {
+  include: ['keywords', 'location'],
+  sort: 'start_time',
+  startDate: '2020-08-11T03',
+  superEvent: superEventId,
+};
+
+const mocks = [
+  {
+    request: {
+      query: EventListDocument,
+      variables,
+    },
+    result: otherEventsResponse,
+  },
+  {
+    request: {
+      query: EventListDocument,
+      variables: { ...variables, page: 2 },
+    },
+    result: otherEventsLoadMoreResponse,
+  },
+];
+
+afterAll(() => {
+  clear();
+});
+
+test('should render other event times', async () => {
+  advanceTo(new Date('2020-08-11'));
+  render(<OtherEventTimesEvents event={mockEvent} />, { mocks });
+
+  const toggleButton = screen.getByRole('button', {
+    name: translations.event.otherTimes.buttonShow,
+  });
+
+  userEvent.click(toggleButton);
+
+  await waitFor(() => {
+    expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
+  });
+
+  otherEventsResponse.data.eventList.data.forEach((event) => {
+    const dateStr = getDateRangeStr({
+      start: event.startTime,
+      end: event.endTime,
+      locale: 'fi',
+      includeTime: true,
+      timeAbbreviation: translations.commons.timeAbbreviation,
+    });
+    expect(screen.getByText(dateStr)).toBeInTheDocument();
+  });
+
+  await waitFor(() => {
```

Comment:
I think there is no point in having other stuff than

`expect(screen.getByText(dateStr)).toBeInTheDocument();`

inside this waitFor.Because that is the only thing that you are waiting for. Everything inside this will possibly be executed many times

Reply:
Very good point. I actually changed this so that I'm only waiting loading-spinner to disappear just to make it little bit more readable

Reply:
Great!